graph_creator/models.py

```diff
@@ -3,23 +3,124 @@
+    def timeline(cls):
+        """
+        Provide timeline in days for plotting on x axis.
+
+        Future: add weeks, or month for dynamic range on plots.
+        """
+        timeline_datetimes = cls.objects.order_by(
+            'data_update'
+        ).values_list('data_update', flat=True).distinct()
+
+        timeline_dates = [x.date().strftime('%Y-%m-%d') for x in timeline_datetimes]
+        return timeline_dates
+
+    @classmethod
+    def data_per_period(cls):
+        """
+        Provide total students, courses and instances, from all services per period, day by default.
+
+        We summarize values per day, because in same day we can receive data from multiple different instances.
+        We suppose, that every instance send data only once per day.
+
+        Future: add weeks, month for dynamic range on plots.
+
+        PostgreSQL: extra(select = {"month": '''DATE_TRUNC('month', creation_date)'''}) 
+        SQlite3: extra(select={'date': 'django_date_trunc("day", "testapp_log"."datetime")'})
+        """
+
+        subquery = cls.objects.order_by('data_update').annotate(
+            date_in_days=Trunc('data_update', 'day', output_field=DateField())
+        ).values('date_in_days')
+
+        students_per_day = subquery.annotate(
+            students=Sum('active_students_amount_day')
+        ).values_list('students', flat=True)
+
+        courses_per_day = subquery.annotate(courses=Sum('courses_amount')).values_list('courses', flat=True)
+        instances_per_day = subquery.annotate(instances=Count('secret_token')).values_list('instances', flat=True)
+
+        return list(students_per_day), list(courses_per_day), list(instances_per_day)
+
+    @classmethod
+    def overall_counts(cls):
+        """
+        Provide total count of all instances, courses and students from all instances per previous calendar day.
+        """
+
+        start_of_day, end_of_day = get_previous_day_start_and_end_dates()
+
+        all_unique_instances = DataStorage.objects.filter(data_update__gt=start_of_day, data_update__lt=end_of_day)
+
+        instances_count = all_unique_instances.count()
+        courses_count = all_unique_instances.aggregate(Sum('courses_amount'))['courses_amount__sum']
+        students_count = all_unique_instances.aggregate(
+            Sum('active_students_amount_day'))['active_students_amount_day__sum']
+
+        return instances_count, courses_count, students_count
+
+    @classmethod
+    def worlds_students_per_country_statistics(cls):
+        """
+        Total of students amount per country to display on world map from all instances per previous calendar day.
+
+        Returns:
+            world_students_per_country (dict): Country-count accordance as pair of key-value.
+        """
+
+        start_of_day, end_of_day = get_previous_day_start_and_end_dates()
+
+        students_per_country_unicodes = list(cls.objects.filter(
+            data_update__gt=start_of_day, data_update__lt=end_of_day
```

Comment:
Same thing about `__gte` & `__lte`. Be sure to check you're not missing any data.